### DISCUSSION TITLE
Control dependend fan curves

### DISCUSSION THREAD

**Discussion by chrshndl** [Rem0o/FanControl.Releases#2651] [Ideas]
Hi everyone,

Tl;Dr: Create a fan curve that has n fan controls as input and can perform a "sum", "average", or "min" on those values combined.
The input should not be a temperture source but a configured Control that might already bind to a certain curve itself.

I enjoy FanControl very much which is why I would like to suggest a feature that seems to be missing after playing around with the tool for quite a while now.

Imagine you have 3 Fans on the top and back of your case and 2 Fans on the front.
Front blows air into the case, top and back suck the air out of the case.
Now, assuming they all are of the same size and the amount of air is linear with increasing rpm, it would be great to be able to create a new curve to set certain fans in relation to each other.

So for my case assume a curve that has the "sum" of the 3 fans on the back and top as current rpm and I can now bind my two front fans to match this identical summarized rpm.

This would give me the chance to - by incresing the offset for the front fans - have kind of an overpressure to prevent dust from entering the case too easy.

Edit: I would be happy to help with coding as well of course.

**Comment by Rem0o** [comment]
It's called the "mix" fan curve:
https://getfancontrol.com/docs/

![image](https://github.com/user-attachments/assets/75d7c83c-bceb-445b-826d-baef44543384)

**Reply by chrshndl** [reply]
Not exactly, this lets me pic from curves to find the max. I want the sum of  lets say: Fan A, Fan B, Fan C current RPMs as a curve to bind other fans to.
So lets say A,B and C run on 50 rpm each.
I want a curve with the rpm of those summed up to give me in this case 150 rpm.
I want to bind another fan to this curve to be at least as fast as A,B and C combined or (by adding an offset) even faster to create overpressure.

Edit: So not a fan curve based on sensors but a fan curve based on the rpm of N other fans rpms

**Reply by Rem0o** [reply]
Then select the sum function instead of max?  Have you played with it a bit?

**Reply by chrshndl** [reply]
I have, maybe I am just too stupid but I think I cant make my point clear.

I can create a new mix fancurve and add N curves to it to get the max or sum out of them.
Now what I actually want is a curve that hast the rpm of selected fans as inputs.
Then I want the sum of all rpms from the input fans to create a total rpm that I can bind other fans to.

So this is not on sensor level but an abstraction layer higher.
You take fans that already bind to a curve an use their current rpm summed together as rpm for other fans.

What I want to achieve is to have the fans that blow into the case to always have slightly more rpm in total than all fans that blow out of the case together.

**Reply by Rem0o** [reply]
For that you can use sync fan curves in rpm mode. Then sum as many as you want in a mix.  Rpm mode will be a checkbox in the three dot menu of the sync fan curve.

**Reply by chrshndl** [reply]
Oh that sounds good, I will try that, thank you!

**Comment by chrshndl** [comment]
I created a mockup to be a bit more clear on what I wish:
![Screenshot 2024-08-08 170804](https://github.com/user-attachments/assets/05429427-c649-4779-af32-5ecb8c9cd494)
See how my Radiator control binds to a curve that ist the sum of 3 other fans?
This way I can have the radiator blow a bit more air into the case than all fan cases suck out of the case but still have all fans indirectly be dependent on the System Temp
